libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/ChoiceInput.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Licensed under the MIT License.
+// Copyright (c) Microsoft Corporation. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder.Dialogs.Choices;
+using Microsoft.Bot.Schema;
+
+namespace Microsoft.Bot.Builder.Dialogs.Adaptive.Input
+{
+    /// <summary>
+    /// Declarative text input to gather choices from users
+    /// </summary>
+    public class ChoiceInput : InputWrapper<ChoicePrompt, FoundChoice>
+    {
+        public List<Choice> Choices { get; set; }
```

Comment:
>Choices [](start = 28, length = 7)

You should add a ChoicesProperty as well that lets you use a dynamic list of choices pulled from memory. This should support both objects and arrays. Look at the JS code for an example of this.

Reply:
Fixed.